libgalois/src/Threads.cpp

```diff
@@ -28,6 +28,10 @@ KATANA_EXPORT unsigned int activeThreads = 1;
 
 unsigned int
 katana::setActiveThreads(unsigned int num) noexcept {
+  // Reset "burn power"/"busy wait" mode since it might be configured for a
+  // different number of threads than we have after this call. That can cause
+  // crashes.
+  katana::GetThreadPool().beKind();
```

Comment:
@roshandathathri @gurbinder533 @ddn0 Could one of your verify this is OK?

Reply:
I'm not sure. `burnPower()` does call `beKind()` internally. Was that not sufficient? I mean was there some bug that you noticed and this fixed?

Reply:
The bug was that calling `burnPower(8)` followed by `setActiveThreads(4)` would create a crash. Calling `burnPower` again would be safe. But that's not what I'm referring to. This was happening in the pytest tests since they switch thread count and mode between tests.

Reply:
Ok, that makes sense. This looks good to me. 

@amberhassaan FYI